datasquare_test.go

```diff
@@ -7,54 +7,62 @@ import (
 )
 
 func TestNewDataSquare(t *testing.T) {
-	result, err := newDataSquare([][]byte{{1, 2}}, NewDefaultTree)
-	if err != nil {
-		panic(err)
-	}
-	if !reflect.DeepEqual(result.squareRow, [][][]byte{{{1, 2}}}) {
-		t.Errorf("newDataSquare failed for 1x1 square")
-	}
-
-	result, err = newDataSquare([][]byte{{1, 2}, {3, 4}, {5, 6}, {7, 8}}, NewDefaultTree)
-	if err != nil {
-		panic(err)
-	}
-	if !reflect.DeepEqual(result.squareRow, [][][]byte{{{1, 2}, {3, 4}}, {{5, 6}, {7, 8}}}) {
-		t.Errorf("newDataSquare failed for 2x2 square")
-	}
-
-	_, err = newDataSquare([][]byte{{1, 2}, {3, 4}, {5, 6}}, NewDefaultTree)
-	if err == nil {
-		t.Errorf("newDataSquare failed; inconsistent number of chunks accepted")
-	}
-
-	_, err = newDataSquare([][]byte{{1, 2}, {3, 4}, {5, 6}, {7}}, NewDefaultTree)
-	if err == nil {
-		t.Errorf("newDataSquare failed; chunks of unequal size accepted")
-	}
+	t.Run("1x1", func(t *testing.T) {
+		result, err := newDataSquare([][]byte{{1, 2}}, NewDefaultTree)
+		if err != nil {
+			panic(err)
+		}
+		if !reflect.DeepEqual(result.squareRow, [][][]byte{{{1, 2}}}) {
+			t.Errorf("newDataSquare failed for 1x1 square")
+		}
+	})
+	t.Run("2x2", func(t *testing.T) {
+		result, err := newDataSquare([][]byte{{1, 2}, {3, 4}, {5, 6}, {7, 8}}, NewDefaultTree)
+		if err != nil {
+			panic(err)
+		}
+		if !reflect.DeepEqual(result.squareRow, [][][]byte{{{1, 2}, {3, 4}}, {{5, 6}, {7, 8}}}) {
+			t.Errorf("newDataSquare failed for 2x2 square")
+		}
+	})
+	t.Run("InconsistentNumberOfChunks", func(t *testing.T) {
+		_, err := newDataSquare([][]byte{{1, 2}, {3, 4}, {5, 6}}, NewDefaultTree)
+		if err == nil {
+			t.Errorf("newDataSquare failed; inconsistent number of chunks accepted")
+		}
+	})
+	t.Run("UnequalChunkSize", func(t *testing.T) {
+		_, err := newDataSquare([][]byte{{1, 2}, {3, 4}, {5, 6}, {7}}, NewDefaultTree)
+		if err == nil {
+			t.Errorf("newDataSquare failed; chunks of unequal size accepted")
+		}
+	})
```

Comment:
my interpretation of table driven tests would be to use a format similar to https://github.com/celestiaorg/rsmt2d/blob/d771ce746cea1dad73a475a4b2d51a311476b874/rsmt2d_test.go#L15-L27

I think this would allow us to avoid a lot of code repetition and hopefully increase readability. If all of the sub-tests in a given test cannot fit in that format, then perhaps those sub tests belong in their own test.

Reply:
Yeah that makes a lot more sense - thanks! Will see what I can do there